### DISCUSSION TITLE
Correct method template for sending a message

### DISCUSSION THREAD

**Discussion by julianodev** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#109] [General]
![image](https://github.com/user-attachments/assets/e7b636bb-bf3d-4a34-a340-60c5091be7b9)

Could you help me with what is the correct method for what I'm trying to do?

In this case, my template has

A header that can have an image or video.

A body that can have dynamic parameters.

A footer 

And it may have some action buttons.

What is the correct method?

Note: For templates without buttons and headers, it works using SendTextMessageTemplateAsync

Thank you very much, and congratulations!

**Comment by gabrieldwight** [comment]
you can try to use SendGenericMessageAsync method to test any other template message that is not supported in the library but you will have to construct the message template object. I have some other templateMessage functions but they target specific message object types.
